framework/include/base/XFEMInterface.h

```diff
@@ -47,9 +48,11 @@ class XFEMInterface : public ConsoleStreamInterface
    * Constructor
    */
   explicit
-  XFEMInterface(MooseApp & app):
+  XFEMInterface(MooseApp & app, MooseSharedPointer<FEProblem> & fe_problem):
```

Comment:
making it a `const &` might make the argument passing faster (no copy, no reference counting, and shared pointer is a non-trivial datastructure, right?)